### PR TITLE
Add preconf tx

### PR DESCRIPTION
**Pull request by vladimir-trifonov**

### CODE REVIEW

File: core/types/transaction.go [hunk 1]
```diff
@@ -86,6 +87,7 @@ type TxData interface {
 	gasFeeCap() *big.Int
 	value() *big.Int
 	nonce() uint64
+	deadline() *big.Int
```
Comment (failfmi) [inline core/types/transaction.go:89]: this can be `uint64`

Comment (vladimir-trifonov) [reply core/types/transaction.go:89]: are you sure, I started with uint64, but saw that the block_number is the same (*big.Int) type?

File: core/types/transaction.go [hunk 2]
```diff
@@ -453,6 +455,14 @@ func (tx *Transaction) WithoutBlobTxSidecar() *Transaction {
 	return cpy
 }
 
+// BlobGas returns the blob gas limit of the transaction for blob transactions, 0 otherwise.
```
Comment (failfmi) [inline core/types/transaction.go]: nit: function comment

File: core/types/transaction.go [hunk 3]
```diff
@@ -453,6 +455,14 @@ func (tx *Transaction) WithoutBlobTxSidecar() *Transaction {
...
+func (tx *Transaction) Deadline() *big.Int {
+	if inclusionPreconfirmationTx, ok := tx.inner.(*InclusionPreconfirmationTx); ok {
+		return inclusionPreconfirmationTx.deadline()
+	}
+	return new(uint256.Int).SetUint64(uint64(0)).ToBig()
```
Comment (failfmi) [inline core/types/transaction.go]: return `big.NewInt(0)`

File: core/types/tx_inclusion_preconfirmation.go [hunk 4]
```diff
@@ -34,16 +37,18 @@ type InclusionPreconfirmationTx struct {
 	Value      *big.Int        // wei amount
 	Data       []byte          // contract invocation input data
 	AccessList AccessList      // EIP-2930 access list
+	Deadline   *uint256.Int    // check if the deadline has passed before committing to it
```
Comment (failfmi) [inline core/types/tx_inclusion_preconfirmation.go]: shouldn't this be `*big.Int`?

File: core/types/tx_inclusion_preconfirmation.go [hunk 5]
```diff
@@ -107,3 +122,19 @@ func (tx *InclusionPreconfirmationTx) encode(b *bytes.Buffer) error {
 func (tx *InclusionPreconfirmationTx) decode(input []byte) error {
 	return rlp.DecodeBytes(input, tx)
 }
+
+func (tx *DynamicFeeTx) deadline() *big.Int {
+	return new(uint256.Int).SetUint64(uint64(0)).ToBig()
+}
+
+func (tx *LegacyTx) deadline() *big.Int {
+	return new(uint256.Int).SetUint64(uint64(0)).ToBig()
```
Comment (failfmi) [inline core/types/tx_inclusion_preconfirmation.go]: `big.NewInt(0)`

File: core/types/tx_inclusion_preconfirmation.go [hunk 6]
```diff
@@ -107,3 +122,19 @@ func (tx *InclusionPreconfirmationTx) encode(b *bytes.Buffer) error {
...
+}
+
+func (tx *AccessListTx) deadline() *big.Int {
+	return new(uint256.Int).SetUint64(uint64(0)).ToBig()
```
Comment (failfmi) [inline core/types/tx_inclusion_preconfirmation.go]: big.NewInt(0)

File: core/types/tx_inclusion_preconfirmation.go [hunk 7]
```diff
@@ -107,3 +122,19 @@ func (tx *InclusionPreconfirmationTx) encode(b *bytes.Buffer) error {
...
+}
+
+func (tx *BlobTx) deadline() *big.Int {
+	return new(uint256.Int).SetUint64(uint64(0)).ToBig()
```
Comment (failfmi) [inline core/types/tx_inclusion_preconfirmation.go]: big.NewInt(0)

**Review by failfmi — APPROVED** [review]

**Pull request merged**